superset/charts/api.py

```diff
@@ -591,25 +603,36 @@ def cache_screenshot(self, pk: int, **kwargs: Any) -> WerkzeugResponse:
 
         chart_url = get_url_path("Superset.slice", slice_id=chart.id)
         screenshot_obj = ChartScreenshot(chart_url, chart.digest)
-        cache_key = screenshot_obj.cache_key(window_size, thumb_size)
+        cache_key = screenshot_obj.get_cache_key(window_size, thumb_size)
+        cache_payload = (
+            screenshot_obj.get_from_cache_key(cache_key) or ScreenshotCachePayload()
+        )
         image_url = get_url_path(
             "ChartRestApi.screenshot", pk=chart.id, digest=cache_key
         )
 
-        def trigger_celery() -> WerkzeugResponse:
+        def build_response(status_code: int) -> WerkzeugResponse:
+            return self.response(
+                status_code,
+                cache_key=cache_key,
+                chart_url=chart_url,
+                image_url=image_url,
+                task_updated_at=cache_payload.get_timestamp(),
+                task_status=cache_payload.get_status(),
+            )
+
+        if cache_payload.should_trigger_task() or force:
```

Comment:
very very nit, but flipping this around will likely save some CPU. To DRY it even more, we could maybe even consider making `force` an optional parameter in `should_trigger_task` that defaults to `False` (not highly opinionated here)
```suggestion
        if force or cache_payload.should_trigger_task():
```